File: algorithms/sorts/bubbleSort/BubbleSort.go
package bubblesort

import "fmt"

func bubbleSort(arr []int) []int {
	sorted := false
	for !sorted {
		sorted = true
		for i := 0; i < len(arr)-1; i++ {
			if arr[i+1] < arr[i] {
				arr[i+1], arr[i] = arr[i], arr[i+1]
				sorted = false
			}
		}
	}
	return arr
}

func Process() {
	arr := []int{22, 1, 4, 7, 12, 55, 2, 551, 242}
	fmt.Println("Before:", arr)
	arr = bubbleSort(arr)
	fmt.Println("After:", arr)
}

File: algorithms/sorts/bubbleSort/BubbleSort.ts
class BubbleSort{
  private process(arr: number[]): number[]{
    let sorted = false
    while(!sorted){
      sorted = true
      for(let i = 0; i < arr.length; i++){
        if(arr[i + 1] < arr[i]){
          let temp = arr[i + 1]
          arr[i + 1] = arr[i]
          arr[i] = temp
          sorted = false
        }
      }
    }
    return arr
  }

  public main(){
    let arr: number[] = [22,33,12,55,23,4,12,53,666,21]
    console.log("Before:", arr)
    arr = this.process(arr)
    console.log("After:", arr)
  }
}

(function(){
  new BubbleSort().main()
})()
